models/ed/R/SAS.ED2.R

```diff
@@ -0,0 +1,502 @@
+  # slow_C_loss <- kcCday_2_umols * A_decomp * decay_rate_ssc * slow_soil_C
+  #---------------------------------------
+  
+  # -----------------------
+  # Calculate the annual carbon loss if things are stable
+  # -----------
+  fsc_loss <- decay_rate_fsc
+  ssc_loss <- decay_rate_ssc
+  ssl_loss <- decay_rate_stsc
+  # -----------
+  
+  
+  # *************************************
+  # Calculate A_decomp according to your DECOMP_SCPEME
+  # A_decomp <- temperature_limitation * water_limitation # aka het_resp_weight
+  # *************************************
+  # ========================
+  # Temperature Limitation 
+  # ========================
+  # soil_tempk <- sum(soil_tempo_y*area.dist)
+  if(decomp_scheme %in% c(0, 3)){
+    temperature_limitation = min(1,exp(resp_temperature_increase * (soil_tempk-318.15)))
+  } else if(decomp_scheme %in% c(1,4)){
+    lnexplloyd             = rh_lloyd_1 * ( rh_lloyd_2 - 1. / (soil_tempk - rh_lloyd_3))
+    lnexplloyd             = max(-38.,min(38,lnexplloyd))
+    temperature_limitation = min( 1.0, resp_temperature_increase * exp(lnexplloyd) )
+  } else if(decomp_scheme==2) {
+    # Low Temp Limitation
+    lnexplow <- rh_decay_low * (rh_low_temp - soil_tempk)
+    lnexplow <- max(-38, min(38, lnexplow))
+    tlow_fun <- 1 + exp(lnexplow)
+    
+    # High Temp Limitation
+    lnexphigh <- rh_decay_high*(soil_tempk - rh_high_temp)
+    lnexphigh <- max(-38, min(38, lnexphigh))
+    thigh_fun <- 1 + exp(lnexphigh)
+    
+    temperature_limitation <- 1/(tlow_fun*thigh_fun)
+  } 
+  # ========================
+  
+  # ========================
+  # Moisture Limitation 
+  # ========================
+  # rel_soil_moist <- sum(swc_y*area.dist)
+  if(decomp_scheme %in% c(0,1)){
+    if (rel_soil_moist <= resp_opt_water){
+      water_limitation = exp( (rel_soil_moist - resp_opt_water) * resp_water_below_opt)
+    } else {
+      water_limitation = exp( (resp_opt_water - rel_soil_moist) * resp_water_above_opt)
+    }
+  } else if(decomp_scheme==2){
+    # Dry soil Limitation
+    lnexpdry <- rh_decay_dry * (rh_dry_smoist - rel_soil_moist)
+    lnexpdry <- max(-38, min(38, lnexpdry))
+    smdry_fun <- 1+exp(lnexpdry)
+    
+    # Wet Soil limitation
+    lnexpwet <- rh_decay_wet * (rel_soil_moist - rh_wet_smoist)
+    lnexpwet <- max(-38, min(38, lnexpwet))
+    smwet_fun <- 1+exp(lnexpwet)
+    
+    water_limitation <- 1/(smdry_fun * smwet_fun)
+  } else {
+    water_limitation = rel_soil_moist * 4.0893 - rel_soil_moist**2 * 3.1681 - 0.3195897
+  }
+  # ========================
+  
+  A_decomp <- temperature_limitation * water_limitation # aka het_resp_weight
+  # *************************************
+  
+  # *************************************
+  # Calculate the steady-state pools
+  # NOTE: Current implementation weights carbon input by patch size rather than using the 
+  #       carbon balance from the oldest state (as was the first implementation)
```

Comment:
Not sure I follow the reason for switching to using a weighted average. Presumably you've also done a weighted average of outputs too not just inputs?  Does this give a result that's very different from the original and/or one that goes to steady state more quickly after the accelerated spin up?